rts/motoko-rts/src/text.rs

```diff
@@ -172,7 +174,7 @@ unsafe extern "C" fn text_to_buf(mut s: Value, mut buf: *mut u8) {
 unsafe extern "C" fn stream_write_text(stream: *mut Stream, mut s: Value) {
     loop {
         let s_ptr = s.as_obj();
-        if s_ptr.tag() == TAG_BLOB {
+        if s_ptr.tag() == TAG_BLOB_B || s_ptr.tag() == TAG_BLOB_T {
```

Comment:
Just curious: The `TAG_BLOB_B` would occur for the Candid streaming. And the `TAG_BLOB_T` because of the principal to text conversions?

Reply:
It's because Stream.write_text is is abused to write both the binary header (type table) and textual string contents.